test/instance.ts

```diff
@@ -1644,12 +1645,31 @@ describe('Instance', () => {
       assert.strictEqual(backup.calledWith_[1], BACKUP_NAME);
     });
 
-    it('should return an instance of Copy Backup', () => {
-      const copy_backup = instance.copyBackup(COPY_BACKUP_NAME, BACKUP_NAME) as {} as FakeBackup;
-      assert(copy_backup instanceof FakeBackup);
-      assert.strictEqual(copy_backup.calledWith_[0], instance);
-      assert.strictEqual(copy_backup.calledWith_[1], COPY_BACKUP_NAME);
-      assert.strictEqual(copy_backup.calledWith_[2], BACKUP_NAME);
+    it('should return an instance of Copy Backup', done => {
+      const OPTIONS = {
+      a: 'b',
+    } as CopyBackupOptions;
+      const gaxOpts = {
+        timeout: 1000,
+      };
+      const options = {a: 'b', gaxOptions: gaxOpts};
+
+      const expectedReqOpts = extend({}, OPTIONS, {
+        parent: instance.formattedName_,
+        backupId: COPY_BACKUP_NAME,
+        sourceBackup: BACKUP_NAME,
+      });
+
+        instance.request = config => {
+        assert.strictEqual(config.client, 'DatabaseAdminClient');
+        assert.strictEqual(config.method, 'createBackup');
+        assert.deepStrictEqual(config.reqOpts, expectedReqOpts);
+        assert.notStrictEqual(config.reqOpts, OPTIONS);
+        assert.deepStrictEqual(config.gaxOpts, options.gaxOptions);
+        done();
+      };
+
+      instance.copyBackup(BACKUP_NAME, COPY_BACKUP_NAME, options, assert.ifError);
```

Comment:
I know what the problem is but I don't know enough about the fakes to recommend a solution. The issue is that all calls to backups get intercepted by the FakeBackup class. That class doesn't have a create() method which is why this error is thrown.

So ideally, you would get FakeBackup to return something meaningful when create() is called. You don't want it to actually call DatabaseAdminClient.createBackup RPC because then it's no longer a unit test.

Reply:
That makes sense Shanika. Let me try to maybe review the PR where these fake backup's were as that may have the way to set it all up.